api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java

```diff
@@ -36,6 +36,7 @@
   private final String name;
 
   public static TableIdentifier of(String... names) {
+    Preconditions.checkArgument(names != null, "Cannot create table identifier from null array");
```

Comment:
I don't think null check is needed for java varargs. The names will be empty array if no input is given. I verified it locally:

```
private void met(String... strs) {
    System.out.println(Arrays.toString(strs));
  }

met();
```

The result is `[]`

Reply:
The problem isn't when this is used as a varargs call. It is that this actually creates `of(String[] names)` in the class file. So you can call it directly with a `String` array:

```java
String[] levels = null;
return Namespace.of(levels); // fails
```

Reply:
Ah I see, never thought about this use case, thanks!

Reply:
there's also a test that verifies this behavior: https://github.com/apache/iceberg/pull/2703/files#diff-7d459f8dfa23098a11664c5f2399f8a8a6c27d72bfcf4384e9bb580fdaff0963R72